controllers/azurejson_machine_controller.go

```diff
@@ -147,6 +147,10 @@ func (r *AzureJSONMachineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	_, kind := infrav1.GroupVersion.WithKind("AzureCluster").ToAPIVersionAndKind()
 
 	// only look at azure clusters
+	if cluster.Spec.InfrastructureRef == nil {
```

Comment:
Should we add a test case for `InfrastructureRef == nil` to `TestAzureJSONMachineReconciler`?

Reply:
thanks for the nudge, added